packages/simcore-sdk/src/simcore_sdk/node_ports_v2/r_clone.py

```diff
@@ -0,0 +1,212 @@
+            )
+
+        response = await result.json()
+        logger.debug("metadata response %s", response)
+        return response["data"]["entity_tag"]
+
+
+async def _delete_file_meta_data(
+    r_clone_settings: RCloneSettings, s3_object: str, user_id: int
+) -> None:
+    async with _get_client_session(r_clone_settings) as session:
+        url = "{endpoint}/v0/locations/0/files/{s3_object}/metadata".format(
+            endpoint=r_clone_settings.storage_endpoint,
+            s3_object=urllib.parse.quote_plus(s3_object),
+        )
+        logger.debug("%s", f"{url=}")
+        result = await session.delete(url, params=dict(user_id=user_id))
+        if result.status != web.HTTPOk.status_code:
+            raise RCloneError(
+                f"Could not fetch metadata: status={result.status} {await result.text()}"
+            )
+
+
+async def sync_local_to_s3(
+    r_clone_settings: Optional[RCloneSettings],
+    s3_object: str,
+    local_file_path: Path,
+    user_id: int,
+) -> ETag:
+    if r_clone_settings is None:
+        raise RCloneError(
+            (
+                f"Could not sync {local_file_path=} to {s3_object=}, provided "
+                f"config is invalid{r_clone_settings=}"
+            )
+        )
+
+    s3_link = await _get_s3_link(
+        r_clone_settings=r_clone_settings, s3_object=s3_object, user_id=user_id
+    )
+    s3_path = re.sub(r"^s3://", "", s3_link)
+    logger.debug(" %s; %s", f"{s3_link=}", f"{s3_path=}")
+
+    r_clone_config_file_content = get_r_clone_config(r_clone_settings)
+    async with _config_file(r_clone_config_file_content) as config_file_name:
+        source_path = local_file_path
+        destination_path = Path(s3_path)
+        assert local_file_path.name == destination_path.name
+        file_name = local_file_path.name
+
+        # rclone only acts upon directories, so to target a specific file
+        # we must run the command from the file's directory. See below
+        # example for further details:
+        #
+        # local_file_path=`/tmp/pytest-of-silenthk/pytest-80/test_sync_local_to_s30/filee3e70682-c209-4cac-a29f-6fbed82c07cd.txt`
+        # s3_path=`simcore/00000000-0000-0000-0000-000000000001/00000000-0000-0000-0000-000000000002/filee3e70682-c209-4cac-a29f-6fbed82c07cd.txt`
+        #
+        # rclone
+        #   --config
+        #   /tmp/tmpd_1rtmss
+        #   sync
+        #   '/tmp/pytest-of-silenthk/pytest-80/test_sync_local_to_s30'
+        #   'dst:simcore/00000000-0000-0000-0000-000000000001/00000000-0000-0000-0000-000000000002'
+        #   --progress
+        #   --copy-links
+        #   --include
+        #   'filee3e70682-c209-4cac-a29f-6fbed82c07cd.txt'
+        r_clone_command = [
+            "rclone",
+            "--config",
+            config_file_name,
+            "sync",
+            f"'{source_path.parent}'",
+            f"'dst:{destination_path.parent}'",
+            "--progress",
```

Comment:
the "old" way nodeports work it was using tqdm to show progress, at least in the logs. How is this handled here?

Reply:
this progress just goes printed in the stdout of the command since rclone is used as an external tool.